src/sentry/incidents/endpoints/organization_alert_rule_index.py

```diff
@@ -118,8 +119,15 @@ def create_metric_alert(self, request, organization, project=None):
                 }
                 find_channel_id_for_alert_rule.apply_async(kwargs=task_args)
                 return Response({"uuid": client.uuid}, status=202)
+                # TODO handle slack channel lookup AND anomaly detection
             else:
                 alert_rule = serializer.save()
+                if alert_rule.detection_type == AlertRuleDetectionType.DYNAMIC.value:
+                    resp = send_historical_data_to_seer(rule=alert_rule, user=request.user)
+                    if resp.status != 200:
+                        alert_rule.delete()
```

Comment:
Why delete the alert rule if the call fails???

Reply:
I discussed the best way to handle this with Josh - if the call fails that means we now have an alert rule that can't fire because we haven't send the historical data needed to make predictions about anomalies to Seer. It's a little awkward but Seer needs to store it along with the rule ID so I can't do it before it's saved.

Reply:
i think there's basically a data normalization thing going on deeper with seer (it seems like it needs a snuba query id, but it's requiring the alert stuff). buuutttt considering that's likely a lot to refactor and would probably need updates in seer, this seemed like an explicit way to say "if we can't create a seer data set for the alert, we can't create the alert"

Reply:
👍 ah i see yea, this is the `create_alert_rule` method?

Do we need to `serializer.save()` above then?
Can we check `serializer.detection_type` and construct the query from the serializer before creating the alert rule?

might be better to do this so we can avoid 2 requests to the db?

Reply:
On seer side we do not need Snuba query id. We do need an identifier for the alert so that future anomaly detection calls can retrieve the historic metric data. Currently, we are reusing the db generated id for that. We can avoid the save-delete-if-fail loop by switching to a GUID as GUIDs can be generated before storing in the table. However, it will need some changes in the alerting system:
- store the GUID in the alert table
- retrieve and pass it to seerwhen triggering anomaly detection.

Reply:
Ok hmmm.

actually i think i understand the need now.
we need the alert_rule.id for seer to be able to associate this backfilled data with the alert rule it will be monitoring.

this flow seems fine then 👍